tests/unit/preprocessor/RandomFourierGaussPreproc_unittest.cc

```diff
@@ -0,0 +1,85 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Author: Nanubala Gnana Sai
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/features/Features.h>
+#include <shogun/kernel/GaussianKernel.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/preprocessor/RandomFourierGaussPreproc.h>
+
+using namespace shogun;
+using namespace random;
+
+class RFGPTest : public ::testing::Test
+{
+public:
+	virtual void SetUp()
+	{
+		SGMatrix<float64_t> mat(num_features, num_vectors);
+		linalg::range_fill(mat, 1.0);
+		auto features = std::make_shared<DenseFeatures<float64_t>>(mat);
+
+		preproc = std::make_shared<RandomFourierGaussPreproc>();
+		preproc->put(kSeed, seed);
+		preproc->set_width(width);
+		preproc->set_dim_output(target_dim);
+		preproc->fit(features);
+	}
+	virtual void TearDown()
+	{
+	}
+
+protected:
+	const int32_t seed = 100;
+	const index_t num_vectors = 5;
+	const index_t num_features = 3;
+	const index_t target_dim = 400;
+	const float64_t width = 200;
```

Comment:
this is too large. Pls pick something that is somewhat close to 1

Reply:
This width parameter is stored as ```m_log_width = std::log(width/2.0) / 2.0;``` Which, in this case should come around as ~ 2.3025.

Reply:
we would want something close to one in the divisor of exp( -(x-y)^2  / width)